shop/src/main/java/ro/msg/learning/shop/configuration/OrderStrategyConfiguration.java

```diff
@@ -0,0 +1,430 @@
+package ro.msg.learning.shop.configuration;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectReader;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.configurationprocessor.json.JSONArray;
+import org.springframework.boot.configurationprocessor.json.JSONException;
+import org.springframework.boot.configurationprocessor.json.JSONObject;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.web.client.RestTemplate;
+import ro.msg.learning.shop.DTOs.LocationOrderDTOs.LocationDirectionsMatrixAPI;
+import ro.msg.learning.shop.DTOs.LocationOrderDTOs.SimplifiedLocationIdAndAddress;
+import ro.msg.learning.shop.DTOs.LocationOrderDTOs.SimplifiedLocationIdAndDistance;
+import ro.msg.learning.shop.DTOs.orderDto.OrderDTOInput;
+import ro.msg.learning.shop.DTOs.orderDto.OrderDTOOutput;
+import ro.msg.learning.shop.DTOs.orderDto.ProcessedOrderProduct;
+import ro.msg.learning.shop.DTOs.orderDto.SimpleProductQuantity;
+import ro.msg.learning.shop.Entities.*;
+import ro.msg.learning.shop.Repositories.*;
+import ro.msg.learning.shop.exceptions.OrderPlacingException;
+import ro.msg.learning.shop.mappers.LocationMapper;
+import ro.msg.learning.shop.mappers.OrderMapper;
+
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Configuration
+@RequiredArgsConstructor
+public class OrderStrategyConfiguration {
+    private final StockRepository stockRepository;
```

Comment:
OrderStrategyConfiguration should be for configuration only. You could extract different location strategies to different classes and set here in the configuration based on "selectedStrategy" which will be used